autode/transition_states/truncation.py

```diff
@@ -140,12 +147,54 @@ def add_remaining_bonds(truncated_graph, full_graph):
         if (i, j) in truncated_graph.edges:
             continue
 
+        # Don't alter bonding if the atom has changed e.g. C -> H
+        if any(truncated_graph.nodes[k]['atom_label'] != full_graph.nodes[k]['atom_label'] for k in (i, j)):
+            continue
+
         # an edge doesn't exist between atoms i ang j - make it
         truncated_graph.add_edge(i, j)
 
     return None
 
 
+def add_remaining_atoms(truncated_graph, full_graph, s_molecule):
+    """Truncation can lead to a split across a C-C bond in a ring where one
+    of the carbons is no longer has 4 nearest neighbours"""
+
+    for i in deepcopy(truncated_graph.nodes):
+
+        # No modification needed if the valency of this atom is retained
+        n_truncated_neighbours = len(list(truncated_graph.neighbors(i)))
+        n_full_neighbours = len(list(full_graph.neighbors(i)))
+
+        if n_truncated_neighbours == n_full_neighbours:
+            continue
+
+        # Only consider non-swapped atoms e.g. not where C -> H
+        if truncated_graph.nodes[i]['atom_label'] != full_graph.nodes[i]['atom_label']:
+            continue
+
+        logger.warning(f'Atom {i} changed valency in truncation')
+        for n in nx.neighbors(full_graph, i):
+
+            if (i, n) in truncated_graph.edges:
+                continue
+
+            # Missing atom n from the truncated graph - probably truncated
+            # X -> H but was also bonded to another atom also in the truncated
+            # graph
+            x, y, z = s_molecule.atoms[n].coord
+            s_molecule.atoms.append(Atom(atomic_symbol='X', x=x, y=y, z=z))
+
+            # Add the capping H atom in place of the X atom just added
+            # will be the last atom index, if it's just been added
+            add_capping_atom(atom_index=i,
```

Comment:
i'm concerned this will end up with atoms very close to each other in the truncated complex?

Reply:
I know what you mean - seems okay for 5 rings tho. and that’s the closest they could be(?)